### PR TITLE
Issue 426 postgres ssl

### PR DESCRIPTION
**Pull request by nickelbob**
This is in response to this issue - https://github.com/go-spatial/tegola/issues/426.

Thanks to @ear7h for the tip that got me started.

### CODE REVIEW

File: provider/postgis/postgis.go [hunk 1]
```diff
@@ -256,6 +274,68 @@ func NewTileProvider(config dict.Dicter) (provider.Tiler, error) {
 	return p, nil
 }
 
+func configTLS(sslMode string, sslKey string, sslCert string, sslRootCert string, cc *pgx.ConnConfig) error {
```
Comment (ear7h) [inline provider/postgis/postgis.go]: Please add tests for this function

File: provider/postgis/postgis.go [hunk 2]
```diff
@@ -113,6 +122,11 @@ func NewTileProvider(config dict.Dicter) (provider.Tiler, error) {
 		return nil, err
 	}
 
+	sslmode, err := config.String(ConfigKeySSLMode, nil)
```
[the remark this answers is not in the document]
Comment (nickelbob) [reply provider/postgis/postgis.go]: For this I set the default in the ConfigTLS function to "disable" - is that what you meant?

[the remark this answers is not in the document]
Comment (ear7h) [reply provider/postgis/postgis.go]: Negative, the `(dicter).String()` method takes two parameters, the map key and a pointer to a default value. If the pointer is `nil` that attribute is seen as required. So the line should look like:
```go
sslmode := "disable"
sslmode, err := config.String(ConfigKeySSLMode, &sslmode)
```

File: provider/postgis/postgis.go [hunk 3]
```diff
@@ -256,6 +291,67 @@ func NewTileProvider(config dict.Dicter) (provider.Tiler, error) {
 	return p, nil
 }
 
+func ConfigTLS(sslMode string, sslKey string, sslCert string, sslRootCert string, cc *pgx.ConnConfig) error {
+	// Match libpq default behavior
+	if sslMode == "" {
+		sslMode = "disable"
+	}
```
Comment (ear7h) [inline provider/postgis/postgis.go]: Using the config strategy, this can be removed

**Pull request closed without merging**